doc/HLD_of_SNS_Repair.rst

```diff
@@ -0,0 +1,1622 @@
+
+.. image:: Images/Formula-2.PNG
+
+Steady state rates are subject to the following constraints:
+
+.. .. math:: G_{D}^{} + G_{O}^{} \leq A_{D}^{} \cdot B_{D}^{}
+
+.. .. math:: G_{S}^{} \leq A_{S}^{} \cdot B_{S}^{}
+
+.. .. math:: G_{P}^{} \leq A_{P}^{} \cdot B_{P}^{}
+
+.. image:: Images/Formula-3.png
+
+To reconstruct a failed unit in a parity group, N of its N + K - 1 units, scattered across N\ :sub:`D` - 1 devices have to be read, meaning that to reconstruct a device an N/(N\ :sub:`D` - 1) fraction of used space on every device in the pool has to be read, giving
+
+.. .. math:: \text{MTTR}_{D}^{} = \frac{U \cdot S_{D}^{}}{G_{D}^{}} \cdot \frac{N}{N_{D}^{} - \ 1}
+
+.. image:: Images/Formula-4.png
+
+As a mean time to repair a device. To minimize MTTR\ :sub:`D`, G\ :sub:`D` has to be maximized. From the equations and inequalities above, the maximal possible value of G\ :sub:`D` is obviously
+
+.. .. math:: G_{D}^{} = \min_{}\left( A_{S}^{} \cdot B_{S}^{} \cdot \chi,\ A_{P}^{} \cdot B_{P}^{} \cdot \chi,\ \frac{A_{D}^{} \cdot B_{D}^{}}{1\  + \ 1/N} \right)
+
+.. where :math:`\chi = \frac{N_{S}^{} - F_{S}^{}}{N_{D}^{} - F_{D}^{}}`
+
+.. image:: Images/Formula-5.PNG
+
+Let's substitute vaguely reasonable data: 
+
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| Parameter   | Value  | Unit      | Explanation                                                                                              |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| U           | 1.0    |           |                                                                                                          |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| S\ :sub:`D` | 2.0e12 | bytes     | 2TB drive                                                                                                |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`S` | 4.0e9  | bytes/sec | IB QDR                                                                                                   |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`P` | 8.0e9  | bytes/sec | `check-sum throughput <check-sum-throughput>`__                                                          |
++-------------+--------+-----------+----------------------------------------------------------------------------------------------------------+
+| B\ :sub:`D` | 7.0e7  | bytes/sec | `ST31000640SS <ST31000640SS>`__                                                                          |
```

Comment:
The same for "`ST31000640SS <ST31000640SS>`__"

Reply:
Updated.